osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/infra/networking/Mx382AlarmDecoder.java

```diff
@@ -17,72 +17,104 @@ public class Mx382AlarmDecoder extends AlarmDecoder {
 
   private static final byte EVENT_NOTIFICATION_REQUEST = (byte) 0xC2;
   private static final byte[] WPDU_HEADER =
-      new byte[] {0x00, 0x01, 0x00, 0x67, 0x00, 0x66, 0x00, 0x1b};
+      new byte[] {0x00, 0x01, 0x00, 0x67, 0x00, 0x66, 0x00, 0x1d};
+  private static final byte[] WPDU_HEADER_WITH_DATE =
+      new byte[] {0x00, 0x01, 0x00, 0x67, 0x00, 0x66, 0x00, 0x2a};
   private static final byte[] AMM_FORWARDED_ALARM_VERSION_0 =
       new byte[] {(byte) 0x80, (byte) 0x80, (byte) 0x80, (byte) 0x80, (byte) 0x80, (byte) 0x00};
-  private static final byte[] DEVICE_ID_2_CLASS_ID = new byte[] {0x00, 0x01};
-  private static final byte VALUE_ATTRIBUTE_ID = 0x02;
-  private static final byte DSMR22_EQUIPMENT_IDENTIFIER_LENGTH = 0x10;
+  static final byte[] CLASS_ID_1 = new byte[] {0x00, 0x01};
+  static final byte ATTRIBUTE_ID_2 = 0x02;
+
+  public static final int DATA_TYPE_OCTET_STRING = 0x09;
+
+  public static final String EXPECTED_MESSAGE_TEMPLATE =
+      "Expected a %s (%s), but encountered: (%s)";
+
   private final DlmsPushNotification.Builder builder = new DlmsPushNotification.Builder();
 
   public DlmsPushNotification decodeMx382alarm(final InputStream inputStream)
       throws UnrecognizedMessageDataException {
 
-    this.checkBytes(inputStream, WPDU_HEADER, "WPDU-header");
-    this.checkByte(inputStream, EVENT_NOTIFICATION_REQUEST, "event-notification-request");
+    this.handleWpduHeaderBytes(inputStream, "WPDU-header");
+    this.handleByte(inputStream, EVENT_NOTIFICATION_REQUEST, "event-notification-request");
 
-    // Datetime is not used, so can be skipped as well
-    this.skipDateTime(inputStream);
+    // Datetime is read and copied
```

Comment:
The datetime bytes are read but not used

Reply:
Modified the comment that the datetime is skipped.